### ISSUE TITLE
Inspect Referer, User-Agent headers for SQLi, XSS

### ISSUE THREAD

**Issue by lifeforms** [SpiderLabs/owasp-modsecurity-crs#402]
Regularly an application turns up which is vulnerable to exploits in some request headers. Today it was [another popular WordPress plugin](https://www.wordfence.com/blog/2016/07/xss-vulnerability-all-in-one-seo-pack-plugin/).

The thing about referrers and user-agent strings is that they often turn up in logging interfaces. These log viewers are notorious for bad escaping and stored XSS.

I'm somewhat less bullish on SQL injections via User-Agent, but Google is full of results when searching for it, so I'd say let's check those too while we're editing this.

I also note that `ARGS_NAMES` is not checked with detectSQLi in rule 942100, so I think this will miss bare post data.

In short, I feel we should:

- check `REQUEST_HEADERS:User-Agent` 
- check `REQUEST_HEADERS:Referer` 
- check `ARGS_NAMES`

for the SQLi and XSS rules in 941xxx and 942xxx.

**Comment by dune73** [comment]
The vuln scanners definitely go with Referer and UA. Complex applications with dynamic / serialised variable names must be vulnerable to injections in var names.

So I think this makes a lot of sense. Any experience with FPs yet?

**Comment by lifeforms** [comment]
`ARGS_NAMES` was added to detectSQLi in #417.

For 3.0 I'll create a PR to add `REQUEST_HEADERS:User-Agent` and `REQUEST_HEADERS:Referer` to libinjection rules only. This should hopefully at least catch some payloads while keeping the amount of work reasonable.

For 3.1 we can consider checking those headers in (much) more rules, after experience.

**Comment by lifeforms** [comment]
Closing in favor of PR #475.
